components/ipfs/ipfs_utils.cc

```diff
@@ -460,47 +464,110 @@ std::string GetRegistryDomainFromIPNS(const GURL& url) {
       cid, net::registry_controlled_domains::INCLUDE_PRIVATE_REGISTRIES);
 }
 
-// gateway.io/ipfs/bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfy ->
-// ipfs://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfy
-// bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfy.ipfs.gateway.io ->
-// ipfs://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfy
-absl::optional<GURL> TranslateToCurrentGatewayUrl(const GURL& url) {
-  if (!url.is_valid() || !url.SchemeIsHTTPOrHTTPS()) {
-    return absl::nullopt;
+std::string DecodeSingleLabelForm(const std::string& input) {
+  if (input.find('.') != std::string::npos) {
+    return input;
+  }
+  std::vector<char> buff;
+  const char* chars = input.c_str();
+  size_t i = 0;
+  for (i = 0; i < input.size(); i++) {
+    bool is_last = i == input.size() - 1;
+    if (chars[i] == '-' && (is_last || chars[i + 1] != '-')) {
+      buff.push_back('.');
+    } else if (!is_last && chars[i] == '-' && chars[i + 1] == '-') {
+      buff.push_back('-');
+      i++;
+    } else {
+      buff.push_back(chars[i]);
+    }
   }
```

Comment:
Is there some public link specifying this algorithm? Would be nice to have it in comments.

Also it looks like loop body could be simplified to:
```
if (chars[i] == '-' && (i < input.size() - 1) && chars[i + 1] == '-') {
  buff.push_back('-');
  i++;
} else if (chars[i] == '-') {
  buff.push_back('.');
} else {
  buff.push_back(chars[i]);
}
```